internal/testintegration/setup.go

```diff
@@ -110,85 +107,140 @@ func EnvTestSetup() (func(), error) {
 		},
 	}
 
-	teardownFunc := func() {
-		cancel()
-		err := testEnv.Stop()
-		if err != nil {
-			Log.Error(err, "unable to stop envtest environment %v")
-		}
-	}
-
+	// Start the testenv
 	cfg, err := testEnv.Start()
+	th := &EnvTestHarness{
+		testEnvCtx:    ctx,
+		testEnv:       testEnv,
+		testEnvCancel: cancel,
+		cfg:           cfg,
+		cancel:        func() {},
+	}
 	if err != nil {
-		return nil, fmt.Errorf("unable to start kuberenetes envtest %v", err)
+		return th, fmt.Errorf("unable to start kuberenetes envtest %v", err)
```

Comment:
We should return `nil` for the happy path value as a general rule.

Reply:
Ditto below.

Reply:
Fixed. It now calls `th.Teardown()` before returning nil, err.